lib/sample_uploader/sample_uploaderImpl.py

```diff
@@ -642,6 +644,90 @@ def batch_link_samples(self, ctx, params):
                              'output is not type dict as required.')
         # return the results
         return [output]
+
+    def filter_samplesets(self, ctx, params):
+        """
+        :param params: instance of type "FilterSampleSetsParams" ->
+           structure: parameter "workspace_name" of String, parameter
+           "workspace_id" of String, parameter "out_sample_set_name" of
+           String, parameter "sample_set_ref" of list of String, parameter
+           "filter_conditions" of list of type "FilterCondition" (Filter
+           SampleSets) -> structure: parameter "column" of String, parameter
+           "comparison" of String, parameter "value" of String, parameter
+           "condition" of String
+        :returns: instance of type "FilterSampleSetsOutput" -> structure:
+           parameter "report_name" of String, parameter "report_ref" of String
+        """
+        # ctx is the context object
+        # return variables are: output
+        #BEGIN filter_samplesets
+        samples = []
+        for sample_set in self.dfu.get_objects({'object_refs': params['sample_set_ref']})['data']:
+            samples.extend(sample_set['data']['samples'])
+        sample_ids = [{'id': sample['id'], 'version':sample['version']} for sample in samples]
+
+        filter_conditions = []
+        for condition in params['filter_conditions']:
+            values = []
```

Comment:
isn't strictly necessary, as `values` gets defined in the if/else statement below. But I kinda prefer seeing a definition...

Reply:
I also prefer it, also helps prevent issues if any new conditions are introduced IMO